pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -214,13 +244,46 @@ func (c *component) containers() []corev1.Container {
 	}
 	containers = append(containers, proxy)
 
-	collector := corev1.Container{
-		Name:         L7CollectorContainerName,
-		Image:        c.config.collectorImage,
-		Env:          c.collectorEnv(),
-		VolumeMounts: c.collectorVolMounts(),
+	if c.config.LogsEnabled {
+		// Log collection specific container
+		collector := corev1.Container{
+			Name:         L7CollectorContainerName,
+			Image:        c.config.collectorImage,
+			Env:          c.collectorEnv(),
+			VolumeMounts: c.collectorVolMounts(),
+		}
+		containers = append(containers, collector)
+	}
+
+	if c.config.WafEnabled {
+		// Web Application Firewall (WAF) specific container
+		dikastes := corev1.Container{
+			Name:  DikastesContainerName,
+			Image: c.config.dikastesImage,
+			Command: []string{
+				"/dikastes",
+				"server",
+				"--dial", "/var/run/felix/nodeagent/socket",
+				"--listen", "/var/run/dikastes/dikastes.sock",
+				"--rules", "/etc/modsecurity-ruleset",
+			},
+			Env: []corev1.EnvVar{
+				{Name: "LOG_LEVEL", Value: "Info"},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{Name: FelixSync, MountPath: "/var/run/felix"},
```

Comment:
I think in the design we mentioned a need to validate that this is configured properly on FelixConfiguration - this is another one for the backlog / next release. Let's not block this PR on it, but we should revisit how we ensure the system is set up properly.

As it is, a user can turn on WAF without enabling policy sync in Felix and I believe they will get a broken cluster.